diff --git a/node_modules/chromadb/dist/cli.mjs b/node_modules/chromadb/dist/cli.mjs
index 46ef722..421378a 100644
--- a/node_modules/chromadb/dist/cli.mjs
+++ b/node_modules/chromadb/dist/cli.mjs
@@ -30,6 +30,8 @@ if (platform === "darwin") {
 } else if (platform === "win32") {
   if (arch === "arm64") {
     binding = require2("chromadb-js-bindings-win32-arm64-msvc");
+  } else if (arch === "x64") {
+    binding = require2("chromadb-js-bindings-win32-x64-msvc");
   } else {
     throw new Error(
       `Unsupported Windows architecture: ${arch}. Only ARM64 is supported.`
