diff --git a/node_modules/@langchain/community/dist/vectorstores/hnswlib.cjs b/node_modules/@langchain/community/dist/vectorstores/hnswlib.cjs
index 33f77cd..dd35dc4 100644
--- a/node_modules/@langchain/community/dist/vectorstores/hnswlib.cjs
+++ b/node_modules/@langchain/community/dist/vectorstores/hnswlib.cjs
@@ -37,6 +37,7 @@ exports.HNSWLib = void 0;
 const vectorstores_1 = require("@langchain/core/vectorstores");
 const documents_1 = require("@langchain/core/documents");
 const in_memory_js_1 = require("../stores/doc/in_memory.cjs");
+const math_1 = require("@langchain/core/utils/math");
 /**
  * Class that implements a vector store using Hierarchical Navigable Small
  * World (HNSW) graphs. It extends the SaveableVectorStore class and
@@ -101,7 +102,13 @@ class HNSWLib extends vectorstores_1.SaveableVectorStore {
             this.index = await HNSWLib.getHierarchicalNSW(this.args);
         }
         if (!this.index.getCurrentCount()) {
+            console.log('index: init', vectors.length);
             this.index.initIndex(vectors.length);
+        } else {
+            if (this.index.getCurrentCount() === 0) {
+                console.log('index: count: init', vectors.length);
+                this.index.initIndex(vectors.length);
+            }            
         }
     }
     get index() {
@@ -138,6 +145,7 @@ class HNSWLib extends vectorstores_1.SaveableVectorStore {
         }
         const capacity = this.index.getMaxElements();
         const needed = this.index.getCurrentCount() + vectors.length;
+        console.log('addVectors:', needed, capacity);
         if (needed > capacity) {
             this.index.resizeIndex(needed);
         }
@@ -189,6 +197,18 @@ class HNSWLib extends vectorstores_1.SaveableVectorStore {
             result.distances[resultIndex],
         ]);
     }
+
+    async maxMarginalRelevanceSearch(query, options) {
+        const queryEmbedding = await this.embeddings.embedQuery(query);
+        const searches = await this.similaritySearchVectorWithScore(queryEmbedding, options.fetchK ?? 20, options.filter);
+        const embeddingList = searches.map((searchResp) => searchResp[1]);
+        const mmrIndexes = (0, math_1.maximalMarginalRelevance)(queryEmbedding, embeddingList, options.lambda, options.k);
+        return mmrIndexes.map((idx) => new documents_1.Document({
+            metadata: searches[idx][0].metadata,
+            pageContent: searches[idx][0].content,
+            id: searches[idx][0].id,
+        }));
+    }
     /**
      * Method to delete the vector store from a directory. It deletes the
      * hnswlib.index file, the docstore.json file, and the args.json file from
